src/test/java/org/openmrs/contrib/qaframework/automation/ManageUserAccountSteps.java

```diff
@@ -0,0 +1,136 @@
+	public void visitHomePage() {
+		initiateWithLogin();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_USER_ACCOUNT)
+	public void destroy() {
+		quit();
+	}
+
+	@Given("a user clicks on System Administartion app from home page")
+	public void loadSystemAdministrationPage() {
+		systemAdministrationPage = homePage.goToSystemAdministrationPage();
+	}
+
+	@Then("the system loads system administration page")
+	public void systemloadsSystemAdministrationPage() {
+		assertTrue(textExists("System Administration"));
+	}
+
+	@When("a user clicks on Manage accounts app")
+	public void loadManageAccountsPage() {
+		userAccountPage = systemAdministrationPage.goToManageAccounts();
+	}
+
+	@Then("the system loads manage acccount page")
+	public void systemloadsManageAccountsPage() {
+		assertTrue(textExists("Manage Accounts"));
+	}
+
+	@When("a user clicks Add New Account button")
+	public void goToAddNewAccountPage() {
+		userAccountPage.clickOnAddUser();
+	}
+
+	@Then("the system loads add new account form")
+	public void systemLoadsAddNewAccountPage() {
+		assertTrue(textExists("Add New Account"));
+	}
+
+	@When("a user enters Data clerk details in the user account form")
+	public void registerDataClerkDetails() {
+		userAccountPage.enterPersonalDetails("Data", "Clerk");
+		userAccountPage.selectGender();
+		userAccountPage.clickOnAddUserAccount();
+		userAccountPage.enterUsername("sclerk");
+		userAccountPage.setUserPrivilegeLevel("High");
+		userAccountPage.setUserPassword("Clerk123", "Clerk123");
+		assertFalse(userAccountPage.containsText(SYSTEM_ALERT));
+		List<String> validationErrors = userAccountPage.getValidationErrors();
+		assertTrue(userAccountPage.isConfirmPasswordMatching(validationErrors));
+		userAccountPage.selectConfiguresForms();
+		userAccountPage.selectRecordsAllergies();
+		userAccountPage.selectRegistersPatients();
+		userAccountPage.selectUsesPatientSummary();
+		userAccountPage.addProviderDetails();
+		userAccountPage.setUserIdentifier("C010");
+		userAccountPage.setUserProviderRole("Clerk");
+	}
+
+	@When("a user enters Nurse details in the user account form")
+	public void registerNurseDetails() {
+		userAccountPage.enterPersonalDetails("Senior", "Nurse");
+		userAccountPage.selectGender();
+		userAccountPage.clickOnAddUserAccount();
+		userAccountPage.enterUsername("snurse");
+		userAccountPage.setUserPrivilegeLevel("High");
+		userAccountPage.setUserPassword("Nurse123", "Nurse123");
+		assertFalse(userAccountPage.containsText(SYSTEM_ALERT));
+		List<String> validationErrors = userAccountPage.getValidationErrors();
+		assertTrue(userAccountPage.isConfirmPasswordMatching(validationErrors));
+		userAccountPage.selectAdministersSystem();
+		userAccountPage.selectEntersVitals();
+		userAccountPage.selectRegistersPatients();
+		userAccountPage.selectUsesPatientSummary();
+		userAccountPage.addProviderDetails();
+		userAccountPage.setUserIdentifier("N101");
+		userAccountPage.setUserProviderRole("Nurse");
```

Comment:
thanks @kdaud  is this the last step when on registering a nurse/user ?

Reply:
When user details are entered into the user account form, the last step is to click on the saveButton in order to save the user details into the system. Since this is done when registering any user, I opted to make it re-usable [here](https://github.com/openmrs/openmrs-contrib-qaframework/blob/d7fcd11e2d1f2a80ba8436fa72fd5ef68aab3e98/src/test/java/org/openmrs/contrib/qaframework/automation/ManageUserAccountSteps.java#L127) so that its just invoked whenever registering a user!

Reply:
He asked whether its the last step while registering a nurse ! i think.

Reply:
A nurse is a user and the explanation [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/78#discussion_r691765231) is generic that caters for registering either a doctor, or clerk or nurse since the registration form is the same what differs is only the user details!